xmodule/capa_block.py

```diff
@@ -263,11 +289,20 @@ class ProblemBlock(
     )
     correct_map = Dict(help=_("Dictionary with the correctness of current student answers"),
                        scope=Scope.user_state, default={})
+    correct_map_history = List(
+        help=_("List of correctness maps for each attempt"), scope=Scope.user_state, default=[]
+    )
     input_state = Dict(help=_("Dictionary for maintaining the state of inputtypes"), scope=Scope.user_state)
     student_answers = Dict(help=_("Dictionary with the current student responses"), scope=Scope.user_state)
+    student_answers_history = List(
+        help=_("List of student answers for each attempt"), scope=Scope.user_state, default=[]
+    )
 
     # enforce_type is set to False here because this field is saved as a dict in the database.
     score = ScoreField(help=_("Dictionary with the current student score"), scope=Scope.user_state, enforce_type=False)
+    score_history = ListScoreField(
```

Comment:
why do we need a new field for this?

Reply:
Because if it is stored as a `List` field, when querying the value it will have a dictionary, and it should have an instance of `Score`